### PR TITLE
DCS-1506 added middleware to trim user inputs

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/views/pages/unexpectedArrivals/noExistingRecordsFound.njk [hunk 1]
```diff
@@ -30,7 +30,7 @@
             }) }}
         </form>
         <p>
-            <a class="govuk-body govuk-link govuk-link--no-visited-state" href="" data-qa="choose-prisoner">
+            <a class="govuk-body govuk-link govuk-link--no-visited-state" href="/confirm-arrival/choose-prisoner" data-qa="choose-prisoner">
```
Comment (louisenorris) [inline server/views/pages/unexpectedArrivals/noExistingRecordsFound.njk:33]: Added missing href - perhaps this should have been done on a separate bug fix ticket but such a small fix hope this is ok.

File: server/utils/utils.ts [hunk 2]
```diff
@@ -66,6 +66,22 @@ export function assertHasOptionalStringValues<K extends string>(
...
 }
 
+export function trimObjectValues(obj: unknown): Record<string, string> {
+  const isObject = obj && typeof obj === 'object'
+
+  if (!isObject) {
+    throw Error('Not a record')
+  }
+
+  if (Object.values(obj).some(s => typeof s !== 'string' && s !== undefined)) {
```
Comment (GurnankCheema) [inline server/utils/utils.ts:76]: maybe .every, not .some?

File: server/utils/utils.ts [hunk 3]
```diff
@@ -66,6 +66,22 @@ export function assertHasOptionalStringValues<K extends string>(
...
+    throw Error('Values present not all strings')
+  }
+  return Object.keys(obj).reduce((acc, curr) => {
+    acc[curr] = obj[curr].trim()
```
Comment (GurnankCheema) [inline server/utils/utils.ts:80]: looks good

File: server/utils/utils.ts (hunk 2 above)
Comment (GurnankCheema) [reply server/utils/utils.ts:76]: cool

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**